server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -1106,4 +1106,8 @@ public static boolean assertCurrentMethodIsNotCalledRecursively() {
         }
         return true;
     }
+
+    public interface TimeSupplier {
+        long timeInMillis();
+    }
```

Comment:
I don't know what's the best place for this. I've added it here. It could be also in a separate file in the same package, or entirely some where else. Suggestions are welcomed. I've also kept it simple to have one method. It could also have `timeInNanos` but that would be unused.

Reply:
I added it after your suggestion! :)

Reply:
I agree, I think we should disassociate it from the thread pool, and keep the `relative`/`absolute` distinctions as well because I think they're important (i.e. being explicit about whether the returned value has a wall-clock interpretation or not). ThreadPool could just `implements TimeSupplier` then.

Reply:
If we wanted it to be a `@FunctionalInterface` we could just derive the other methods from an `absoluteTimeInMillis` implementation e.g.

absoluteTimeInMillis = ???
relativeTimeInMillis = absoluteTimeInMillis
rawRelativeTimeInMillis = absoluteTimeInMillis
relativeTimeInNanos = absoluteTimeInMillis * 1_000_000

Reply:
I had something in mind like this as a test utility have added it.